gitbook-docs/data_model.md

```diff
@@ -201,7 +202,11 @@ The `updates` holds an array (JSON array) of updates, each of which has a `sourc
 An `update` has a single `source` value and it applies to each of the `values` items.
 In cases where you can get data from only a single source the source may be omitted and the receiver may fill it in when multiplexing data from several sources.
 
-Each 'value' item is then simply a pair of 'relative path', and 'value'.
+Each `value` item is then simply a pair of `path` and `value`.
+The `path` must be a _leaf path_: it must be a path to a leaf the of the full model.
+A leaf is where the actual value of the Signal K property is and where `timestamp`, `$source` and `values` properties are in the full model.
+The value is often a scalar - a numeric value, as in the example above, but it can also be an object.
```

Comment:
The issue is with consistency between the two views of the model. For a given value, both the path used to identify it and its type should be the same in both views.

For example, to read a vessel's heading using the REST API, an application needs to GET from `{api}/{context}/{path}/value` where context is `vessels/{vesselId}` and path is `navigation/heading` whereas updates are sent as a delta where context is `vessels.{vesselId}` and path is `navigation.heading`.

A leaf value can also store `meta` information. If the `value` is the same as the full model (i.e. an `object`) then changes to that (e.g. changes to alarm zones) can be included in a delta. If it is just a scalar then it cannot and a separate mechanism will be needed.